articles/quickstart/webapp/aspnet-core-blazor-server/interactive.md

```diff
@@ -0,0 +1,137 @@
+
+<%= include('../../_includes/_configure_auth0_interactive', { 
+  callback: 'http://localhost:3000/callback',
+  returnTo: 'http://localhost:3000'
+}) %>
+
+## Install and Configure the SDK {{{ data-action=code data-code="Program.cs" }}}
+
+### Install from Nuget
+
+To integrate Auth0 with Blazor Server you can use our SDK by installing the `Auth0.AspNetCore.Authentication` [Nuget package](https://www.nuget.org/packages/Auth0.AspNetCore.Authentication/) to your application.
+
+```bash
+Install-Package Auth0.AspNetCore.Authentication
+```
+
+### Configure the middleware
+
+To enable authentication in your Blazor Server application, use the middleware provided by the SDK. Go to the `Program.cs` file and call `builder.Services.AddAuth0WebAppAuthentication()` to register the SDK's middleware.
+
+Ensure to configure the `Domain` and `ClientId`, these are required fields to ensure the SDK knows which Auth0 tenant and application it should use.
+
+Make sure you have enabled authentication and authorization in your `Program.cs` file.
+
+## Login {{{ data-action=code data-code="Login.cshtml.cs" }}}
+
+To allow users to login to your Blazor Server application, add a `LoginModel` to your `Pages` directory.
+
+Inside the `LoginModel`'s `OnGet` method, call `HttpContext.ChallengeAsync()` and pass `Auth0Constants.AuthenticationScheme` as the authentication scheme. This will invoke the OIDC authentication handler that our SDK registers internally. Be sure to also specify the corresponding `authenticationProperties`, which you can construct using the `LoginAuthenticationPropertiesBuilder`.
+
+After succesfully calling `HttpContext.ChallengeAsync()`, the user will be redirected to Auth0 and signed in to both the OIDC middleware and the cookie middleware upon being redirected back to your application. This will allow the users to be authenticated on subsequent requests.
+
+::::checkpoint
+
+:::checkpoint-default
+
+Now that you have configured Login, run your application to verify that:
+* Navigating to your `Login` page will redirect to Auth0
+* Entering your credentials will redirect you back to your application.
+
+:::
+
+:::checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+* make sure the correct application is selected
+* did you save after entering your URLs?
+* make sure the domain and client ID are configured correctly
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+
+:::
+
+::::
+
+## Display User Profile {{{ data-action=code data-code="Profile.razor" }}}
+
+After the middleware has successfully retrieved the tokens from Auth0, it will extract the user's information and claims from the ID Token and makes them available through the `AuthenticationState`, which you can add as a `CascadingParameter`.
```

Comment:
```suggestion
After the middleware has successfully retrieved the tokens from Auth0, it will extract the user's information and claims from the ID Token and make them available through the `AuthenticationState`, which you can add as a `CascadingParameter`.
```